argo-workflows/nautobot-interface-sync/code/sync_nautobot_interfaces.py

```diff
@@ -0,0 +1,58 @@
+import sushy
+from models import Chassis
+from helpers import *
+
+logger = setup_logger(__name__)
+
+
+def create_or_update_nb_intf(nb_session, nb_device, redfish_intf):
+    inf = nb_session.dcim.interfaces.get(device_id=nb_device.id, name=redfish_intf.name)
+    if inf is None:
+        logger.info(
+            f"{redfish_intf.name} was NOT found for {nb_device.name}, creating..."
+        )
+
+        nb_session.dcim.interfaces.create(
+            device=nb_device.id,
+            name=redfish_intf.name,
+            mac_address=redfish_intf.mac_addr,
+            type="other",
+            status="Active",
+            description=f"Location: {redfish_intf.location}",
+        )
+        logger.info(f"{redfish_intf.name} successfully created")
+    else:
+        logger.info(f"Found {inf.name} for {nb_device.name}")
+        if not inf.mac_address:
+            inf.update({"mac_address": i.mac_addr})
+            logger.info(f"Mac address for {inf.name} successfully updated")
+
```

Comment:
Here's suggested version with minor bugfixes and more descriptive variable names 
```suggestion
def create_or_update_nb_intf(nb_session, nb_device, redfish_interface):
    mac_address = redfish_interface.mac_addr
    interface_name = redfish_interface.name
    nautobot_interface = nb_session.dcim.interfaces.get(
        device_id=nb_device.id,
        name=interface_name
    )
    if nautobot_interface:
        logger.info(f"Found {interface_name} for {nb_device.name}")
        if not nautobot_interface.mac_address:
            nautobot_interface.update({"mac_address": mac_addr})
            logger.info(f"Mac address for {inf.name} successfully updated")
    else:
        logger.info(
            f"{interface_name} was NOT found for {nb_device.name}, creating..."
        )

        nb_session.dcim.interfaces.create(
            device=nb_device.id,
            name=interface_name,
            mac_address=mac_address,
            type="other",
            status="Active",
            description=f"Location: {redfish_interface.location}",
        )
        logger.info(f"{interface_name} successfully created")

```

Reply:
completely reformatted now